File: Demo/Sources/ViewControllers/Transition/Child/TransitionChildViewController.m
//
//  Globus
//

#import "TransitionChildViewController.h"

@implementation TransitionChildViewController

#pragma mark - Actions

- (IBAction)pressedClose:(id)sender {
    [self.navigationController dismissViewControllerAnimated:YES completion:nil];
}

- (IBAction)pressedModal:(id)sender {
    TransitionChildViewController* vc = [TransitionChildViewController instantiate];
    GLBNavigationViewController* nvc = [GLBNavigationViewController viewControllerWithRootViewController:vc];
    nvc.transitionNavigation = self.transition;
    nvc.transitionModal = self.transition;
    [self presentViewController:nvc animated:YES completion:nil];
}

- (IBAction)pressedPush:(id)sender {
    TransitionChildViewController* vc = [TransitionChildViewController instantiate];
    vc.transitionModal = self.transition;
    [self.navigationController pushViewController:vc animated:YES];
}

- (IBAction)pressedPop:(id)sender {
    [self.navigationController popViewControllerAnimated:YES];
}

@end

File: Demo/Sources/ViewControllers/Transition/Child/TransitionChildViewController.swift
//
//  Globus
//

import Globus

class TransitionChildViewController: TransitionViewController {
    
    // MARK - Action
    
    @IBAction internal func pressedClose(_ sender: Any) {
        self.navigationController!.dismiss(animated: true, completion: nil)
    }
    
    @IBAction internal func pressedModal(_ sender: Any) {
        let vc = TransitionChildViewController.instantiate()!
        let nvc = GLBNavigationViewController.init(rootViewController: vc)
        nvc.transitionNavigation = self.transition
        nvc.transitionModal = self.transition
        self.present(nvc, animated: true, completion: nil)
    }
    
    @IBAction internal func pressedPush(_ sender: Any) {
        let vc = TransitionChildViewController.instantiate()!
        vc.transitionModal = self.transition
        self.navigationController!.pushViewController(vc, animated: true)
    }
    
    @IBAction internal func pressedPop(_ sender: Any) {
        _ = self.navigationController!.popViewController(animated: true)
    }
    
    // MARK: - GLBNibExtension
    
    public override static func nibName() -> String {
        return "TransitionChildViewController-Swift"
    }
    
}
